src/snoowrap.js

```diff
@@ -1033,48 +1033,28 @@ const snoowrap = class snoowrap {
   searchSubredditNames ({exact = false, include_nsfw = true, includeNsfw = include_nsfw, query}) {
     return this._post({uri: 'api/search_reddit_names', qs: {exact, include_over_18: includeNsfw, query}}).get('names');
   }
-  _createOrEditSubreddit ({
-    allow_images = true,
-    allow_top = true,
-    captcha,
-    captcha_iden,
-    collapse_deleted_comments = false,
-    comment_score_hide_mins = 0,
-    description,
-    exclude_banned_modqueue = false,
-    'header-title': header_title,
-    hide_ads = false,
-    lang = 'en',
-    link_type = 'any',
-    name,
-    over_18 = false,
-    public_description,
-    public_traffic = false,
-    show_media = false,
-    show_media_preview = true,
-    spam_comments = 'high',
-    spam_links = 'high',
-    spam_selfposts = 'high',
-    spoilers_enabled = false,
-    sr,
-    submit_link_label = '',
-    submit_text_label = '',
-    submit_text = '',
-    suggested_comment_sort = 'confidence',
-    title,
-    type = 'public',
-    wiki_edit_age,
-    wiki_edit_karma,
-    wikimode = 'modonly'
-  }) {
-    return this._post({uri: 'api/site_admin', form: {
-      allow_images, allow_top, api_type, captcha, collapse_deleted_comments, comment_score_hide_mins, description,
-      exclude_banned_modqueue, 'header-title': header_title, hide_ads, iden: captcha_iden, lang, link_type, name,
-      over_18, public_description, public_traffic, show_media, show_media_preview, spam_comments, spam_links,
-      spam_selfposts, spoilers_enabled, sr, submit_link_label, submit_text, submit_text_label, suggested_comment_sort,
-      title, type, wiki_edit_age, wiki_edit_karma, wikimode
-    }}).then(handleJsonErrors(this.getSubreddit(name || sr)));
+
+  _createOrEditSubreddit(form={}) {
+    let safe = ['all_original_content', 'allow_discovery', 'allow_images', 'allow_post_crossposts', 'allow_top', 'allow_videos', 'api_type',
+                'collapse_deleted_comments', 'comment_score_hide_mins', 'description', 'exclude_banned_modqueue', 'free_form_reports', 'g-recaptcha-response ',
+                'header-title', 'hide_ads', 'key_color', 'lang', 'link_type', 'name', 'original_content_tag_enabled', 'over_18', 'public_description',
+                'show_media', 'show_media_preview', 'spam_comments', 'spam_links', 'spam_selfposts', 'spoilers_enabled', 'sr', 'submit_link_label', 'submit_text',
+                'submit_text_label', 'suggested_comment_sort', 'theme_sr', 'theme_sr_update', 'title', 'type', 'uh / X-Modhash header', 'wiki_edit_age',
+                'wiki_edit_karma', 'wikimode'];
```

Comment:
Instead of explicitly listing all the keys here, can we add a "catch-all" for any future keys that are added?

```diff
   wikimode = 'modonly',
+  ...otherKeys
}) {
```

Then the keys could be included in the request like this:

```js
return this._post({
  /* other stuff that's already present, */
  ...otherKeys
```

Reply:
How would the catch all work differently from the argument deconstruction that was previously there?  Essentially, the object being passed into the `_post` method here has a bunch of internal properties it probably shouldn't have on it in addition to what's returned from the actual API.  So it's not enough to just say anything not already recognized should be added, since the API request will fail.

If the argument deconstruction just had the new keys added in and then they were referenced in the form post request, it would work in the manner you're suggesting.

I think that the process should be refactored though so that the values passed into the post message are only the combination of the user-set values in addition to the pre-existing settings coming back from the services.

Reply:
> I think that the process should be refactored though so that the values passed into the post message are only the combination of the user-set values in addition to the pre-existing settings coming back from the services.

Yes, that's what I mean. To clarify, by "other stuff that's already present" I meant the code that's already present, not the keys from reddit that are already present.